### PR TITLE
Update roadmap

### PR DESCRIPTION
**Pull request by GearsDatapacks**
This PR updates the roadmap by moving some of items from the `planned` section to `in-progress`, because they are now being worked on or have been merged in to `main`. It also adds a few new items in the `planned` section, as several new issues have been opened since the roadmap was last updated.

### CODE REVIEW

File: _data/roadmap.yml [hunk 1]
```diff
@@ -14,19 +14,22 @@ in-progress:
...
-  - Language server extract variable code action
-  - Language server inline variable code action
   - Language server label completion
   - Language server renaming
   - Mutual tail call optimisation on JavaScript
   - OTP named process support
   - Test keyword `assert`
   - API key generation when Hex rate limits reached
+  - Fault tolerant parsing of `case` expressions
+  - Optimisations in code generation on the JavaScript target
```
Comment (lpil) [inline _data/roadmap.yml:32]: This seems a little vague- what does it refer to?

Comment (GearsDatapacks) [reply _data/roadmap.yml:32]: It's 3 separate issues regarding various optimisations (namely removing iifes, using a decision tree for case expressions, and lifting constants into global scope)
I can split it into three separate bullet points

Comment (lpil) [reply _data/roadmap.yml:32]: Yes please

File: _data/roadmap.yml [hunk 2]
```diff
@@ -14,19 +14,22 @@ in-progress:
   - Language server convert to/from `use` code action
   - Local Hex API key encryption
   - Unaligned bit array support on JavaScript
+  - Language server extract variable code action
+  - Language server inline variable code action
```
Comment (lpil) [inline _data/roadmap.yml:18]: I don't think this one has been started

Comment (GearsDatapacks) [reply _data/roadmap.yml:18]: Oh sorry, it's the expand function capture one that has. I'll fix that

**Review by lpil — APPROVED** [review]
Thank you!!

**Pull request merged**